swiftpkg/internal/bazel_repo_names.bzl

```diff
@@ -17,6 +17,16 @@ def _from_identity(identity):
     Returns:
         A Bazel repository name as a `string`.
     """
+
+    # TODO: Should we do this replacement for registry packages?
+    # This potentially conflicts with Swift Package Manager's `--use-registry-identity-for-scm` flag.
+    # Without this though, what would have been generated as `swiftpkg_swift_collections` is instead
+    # `swiftpkg_apple.swift_collections` which seems more confusing and build breaking if going from registry
+    # to non-registry.
+    registry_org_identity_split = identity.split(".")
+    if len(registry_org_identity_split) > 1:
+        identity = registry_org_identity_split[1]
+
```

Comment:
Could we run into collision issues if a package has the name that we are replacing into?

Reply:
Yes, but I believe that would be the point of `--use-registry-identity-for-scm/--replace-scm-with-registry` flags in SPM. Since this makes up the package identity I imagine a registry user would want to use these flags always to enforce all dependencies resolve using the registry versions (or fail to find things outside of it).

Maybe what we should do here is:

- If `--replace-scm-with-registry` (via some equivalent Bazel attr) are set in the `swift_deps.configure_swift_package` extension, then we do this replacement. Otherwise we don't.

This means you can end up with potentially these two labels:

```
swftpkg_swift_collections
swiftpk_apple.swift_collection
```

Reply:
Decided against doing this transformation and instead added a `replace_scm_with_registry` attr to the `configure_swift_package` tag which will set the `--replace-scm-with-registry` SPM flag.

This means labels for registries will generally look like:

```
@swiftpkg_[org].[name]
```